src/agent/onefuzz/src/heartbeat.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::utils::CheckNotify;
+use anyhow::Result;
+use reqwest::Url;
+use serde::{self, Deserialize, Serialize};
+use std::{
+    collections::HashSet,
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use storage_queue::QueueClient;
+use tokio::{sync::Notify, task::JoinHandle};
+
+use uuid::Uuid;
+
+use futures::Future;
+
+#[derive(Debug, Deserialize, Serialize, Hash, Eq, PartialEq, Clone)]
+#[serde(tag = "type")]
+pub enum HeartbeatData {
+    TaskAlive,
+    MachineAlive,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone)]
+struct Heartbeat {
+    task_id: Option<Uuid>,
+    machine_id: Uuid,
+    machine_name: String,
+    data: Vec<HeartbeatData>,
```

Comment:
Is it correct that this is a queue of pending/not-yet-flushed heartbeat messages?

Maybe we should name it `pending` instead of the more generic `data`.